src/sagemaker/cli/compatibility/v2/modifiers/serde.py

```diff
@@ -0,0 +1,413 @@
+version 2.0 and later of the SageMaker Python SDK.
+"""
+from __future__ import absolute_import
+
+import ast
+
+import pasta
+
+from sagemaker.cli.compatibility.v2.modifiers import matching
+from sagemaker.cli.compatibility.v2.modifiers.modifier import Modifier
+
+# The values are tuples so that the object can be passed to matching.matches_any.
+OLD_CLASS_NAME_TO_NAMESPACES = {
+    "_CsvSerializer": ("sagemaker.predictor",),
+    "_JsonSerializer": ("sagemaker.predictor",),
+    "_NpySerializer": ("sagemaker.predictor",),
+    "_CsvDeserializer": ("sagemaker.predictor",),
+    "BytesDeserializer": ("sagemaker.predictor",),
+    "StringDeserializer": ("sagemaker.predictor",),
+    "StreamDeserializer": ("sagemaker.predictor",),
+    "_NumpyDeserializer": ("sagemaker.predictor",),
+    "_JsonDeserializer": ("sagemaker.predictor",),
+    "numpy_to_record_serializer": ("sagemaker.amazon.common",),
+    "record_deserializer": ("sagemaker.amazon.common",),
+}
+
+# The values are tuples so that the object can be passed to matching.matches_any.
+NEW_CLASS_NAME_TO_NAMESPACES = {
+    "CSVSerializer": ("sagemaker.serializers",),
+    "JSONSerializer": ("sagemaker.serializers",),
+    "NumpySerializer": ("sagemaker.serializers",),
+    "CSVDeserializer": ("sagemaker.deserializers",),
+    "BytesDeserializer": ("sagemaker.deserializers",),
+    "StringDeserializer": ("sagemaker.deserializers",),
+    "StreamDeserializer": ("sagemaker.deserializers",),
+    "NumpyDeserializer": ("sagemaker.deserializers",),
+    "JSONDeserializer": ("sagemaker.deserializers",),
+    "RecordSerializer ": ("sagemaker.amazon.common",),
+    "RecordDeserializer": ("sagemaker.amazon.common",),
+}
+
+OLD_CLASS_NAME_TO_NEW_CLASS_NAME = {
+    "_CsvSerializer": "CSVSerializer",
+    "_JsonSerializer": "JSONSerializer",
+    "_NpySerializer": "NumpySerializer",
+    "_CsvDeserializer": "CSVDeserializer",
+    "BytesDeserializer": "BytesDeserializer",
+    "StringDeserializer": "StringDeserializer",
+    "StreamDeserializer": "StreamDeserializer",
+    "_NumpyDeserializer": "NumpyDeserializer",
+    "_JsonDeserializer": "JSONDeserializer",
+    "numpy_to_record_serializer": "RecordSerializer",
+    "record_deserializer": "RecordDeserializer",
+}
+
+OLD_OBJECT_NAME_TO_NEW_CLASS_NAME = {
+    "csv_serializer": "CSVSerializer",
+    "json_serializer": "JSONSerializer",
+    "npy_serializer": "NumpySerializer",
+    "csv_deserializer": "CSVDeserializer",
+    "json_deserializer": "JSONDeserializer",
+    "numpy_deserializer": "NumpyDeserializer",
+}
+
+OLD_AMAZON_CLASS_NAMES = set(
+    {
+        class_name
+        for class_name, namespaces in OLD_CLASS_NAME_TO_NAMESPACES.items()
+        if "sagemaker.amazon.common" in namespaces
+    }
+)
+NEW_AMAZON_CLASS_NAMES = set(
+    {
+        class_name
+        for class_name, namespaces in NEW_CLASS_NAME_TO_NAMESPACES.items()
+        if "sagemaker.amazon.common" in namespaces
+    }
+)
```

Comment:
food for thought: it might be easier to hardcode which two classes are the "Amazon" serializers and build out the dictionaries from there. something like:

```python
OLD_CLASS_NAMES_TO_NAMESPACES = {class_name: "sagemaker.amazon.common" for class_name in OLD_AMAZON_CLASS_NAMES}
OLD_CLASS_NAMES_TO_NAMESPACES.update({class_name: "sagemaker.predictor" for class_name in OLD_PREDICTOR_CLASS_NAMES})
```

@metrizable might have thoughts on what's more Pythonic

Reply:
Fixed